code/edge/src/main/java/com/adobe/marketing/mobile/EdgeHitProcessor.java

```diff
@@ -347,7 +345,94 @@ private EdgeEndpoint getEdgeEndpoint(final Map<String, Object> edgeConfiguration
 		);
 
 		final String locationHint = stateCallback != null ? stateCallback.getLocationHint() : null;
-		return new EdgeEndpoint(requestEnvironment, requestDomain, locationHint);
+
+		// Use null fallback value for request without custom path value
+		String customPath = DataReader.optString(requestProperties, EdgeConstants.EventDataKeys.Request.PATH, null);
+
+		return new EdgeEndpoint(requestType, requestEnvironment, requestDomain, customPath, locationHint);
+	}
+
+	/**
+	 * Extracts all the custom request properties to overwrite the default values
+	 * @param event current event for which the request properties are to be extracted
+	 * @return the map of extracted request properties and their custom values
+	 */
+	private Map<String, Object> getRequestProperties(final Event event) {
+		Map<String, Object> requestProperties = new HashMap<>();
+
+		String overwritePath = getCustomRequestPath(event);
+		if (overwritePath != null) {
+			Log.trace(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Got custom path:(%s) for event:(%s), which will overwrite the default interaction request path.",
+				overwritePath,
+				event.getUniqueIdentifier()
+			);
+			requestProperties.put(EdgeConstants.EventDataKeys.Request.PATH, overwritePath);
+		}
+		return requestProperties;
+	}
+
+	/**
+	 * Extracts network request path property to overwrite the default endpoint path value
+	 * @param event current event for which the request path property is to be extracted
+	 * @return the custom path string
+	 */
+	private String getCustomRequestPath(final Event event) {
+		String path = null;
```

Comment:
Use the `DataReader` class here to extract the path. Remember that DataReader will return null if the passed in map is null, so you don't need the additional isNullOrEmpty checks.
```
       Map<String, Object> requestData = DataReader.optTypedMap(Object.class, event.getEventData(),"request", null);
        String path = DataReader.optString(requestData, "path", null);
```